test/e2e_node/kubelet_test.go

```diff
@@ -116,6 +116,62 @@ var _ = framework.KubeDescribe("Kubelet", func() {
 			Expect(err).To(BeNil(), fmt.Sprintf("Error deleting Pod %v", err))
 		})
 	})
+	Context("when scheduling a busybox Pod with hostAliases", func() {
+		podName := "busybox-host-aliases" + string(uuid.NewUUID())
+		expectedHostsFileContent := `# Kubernetes-managed hosts file.
+127.0.0.1	localhost
+::1	localhost ip6-localhost ip6-loopback
+fe00::0	ip6-localnet
+fe00::0	ip6-mcastprefix
+fe00::1	ip6-allnodes
+fe00::2	ip6-allrouters
+%s	%s
+123.45.67.89	foo
+123.45.67.89	foo.bar
+`
+
+		It("it should write entries to /etc/hosts", func() {
+			podClient.CreateSync(&v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: podName,
+				},
+				Spec: v1.PodSpec{
+					// Don't restart the Pod since it is expected to exit
+					RestartPolicy: v1.RestartPolicyNever,
+					Containers: []v1.Container{
+						{
+							Image:   "gcr.io/google_containers/busybox:1.24",
+							Name:    podName,
+							Command: []string{"/bin/sh", "-c", "cat /etc/hosts; sleep 240"},
+						},
+					},
+					HostAliases: []v1.HostAlias{
+						{
+							IP:        "123.45.67.89",
+							Hostnames: []string{"foo", "foo.bar"},
+						},
+					},
+				},
+			})
+
+			Eventually(func() string {
+				pod, err := podClient.Get(podName, metav1.GetOptions{})
+				Expect(err).To(BeNil(), fmt.Sprintf("Error getting Pod %v", err))
+				podIP := pod.Status.PodIP
+
+				expectedHostsFileContent = fmt.Sprintf(expectedHostsFileContent, podIP, podName)
+
+				rc, err := podClient.GetLogs(podName, &v1.PodLogOptions{}).Stream()
+				if err != nil {
+					return ""
```

Comment:
@vishh could you help us understand why in other test cases it's acceptable to return blank string? sorry i'm not very familiar with the test framework used for the node e2e tests.